File: api/app/schemas/bank_account.py
from datetime import datetime

from pydantic import BaseModel, Field


class BankAccountCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    bank_name: str | None = Field(default=None, max_length=120)
    balance: float = Field(default=0, ge=0)
    currency: str = Field(default="USD", min_length=3, max_length=3)


class BankAccountUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=80)
    bank_name: str | None = Field(default=None, max_length=120)
    balance: float | None = Field(default=None, ge=0)
    currency: str | None = Field(default=None, min_length=3, max_length=3)


class BankAccountResponse(BaseModel):
    id: str
    name: str
    bank_name: str | None = None
    balance: float
    currency: str
    created_at: datetime
    updated_at: datetime

File: mobile/lib/features/bank_accounts/models/bank_account.dart
class BankAccount {
  const BankAccount({
    required this.id,
    required this.name,
    required this.bankName,
    required this.balance,
    required this.currency,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String name;
  final String? bankName;
  final double balance;
  final String currency;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory BankAccount.fromJson(Map<String, dynamic> json) {
    return BankAccount(
      id: json['id'] as String,
      name: json['name'] as String,
      bankName: json['bank_name'] as String?,
      balance: (json['balance'] as num).toDouble(),
      currency: json['currency'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }
}
